src/formatDate.js

```diff
@@ -50,7 +50,48 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateToArr = [];
+  const dateObject = {};
+  const newFormatOfDate = [];
+  let newFormatOfDateToReturn = '';
+
+  for (let i = fromFormat.length - 1; i >= 0; i--) {
+    if (i === fromFormat.length - 1) {
```

Comment:
Isn't it better to move the code inside this if out of the for loop. The variables are assigned only once when the loop starts. So, putting if there makes your code harder to read